RenderManager_xbox/DistortionCorrectRenderManager.cpp

```diff
@@ -0,0 +1,760 @@
+// that we need.
+static osvr::clientkit::ClientContext context(
+  "com.osvr.renderManager.distortionCorrectRenderManager");
+
+static osvr::renderkit::RenderManager *render = nullptr;
+
+// Quadric and sphere to use for rendering.
+static GLUquadric *sphere = nullptr;
+
+// X,Y location
+typedef struct {
+  double x;
+  double y;
+} XY;
+
+static std::vector<float> params;  //< Distortion parameters
+static int activeParam = 0;  //< Which parameter are we adjusting?
+
+static std::string osvrGetString(OSVR_ClientContext context, const std::string& path)
+{
+  size_t len;
+  if (osvrClientGetStringParameterLength(context, path.c_str(), &len) == OSVR_RETURN_FAILURE) {
+    std::string msg = std::string("Couldn't get osvr string length for path ") + path;
+    std::cerr << msg << std::endl;
+    throw std::runtime_error(msg);
+  }
+
+  //struct TempBuffer {
+  //    char *buffer;
+  //    TempBuffer(size_t len) { buffer = new char[len + 1]; }
+  //    ~TempBuffer() { delete[] buffer; }
+  //} tempBuffer(len);
+  std::vector<char> tempBuffer(len + 1);
+
+  if (osvrClientGetStringParameter(context, path.c_str(), tempBuffer.data(), len + 1) == OSVR_RETURN_FAILURE) {
+    std::string msg = std::string("Couldn't get osvr string buffer for path ") + path;
+    std::cerr << msg << std::endl;
+    throw std::runtime_error(msg);
+  }
+  return std::string(tempBuffer.data(), len);
+}
+
+#ifdef _WIN32
+// Note: On Windows, this runs in a different thread from
+// the main application.
+static BOOL CtrlHandler(DWORD fdwCtrlType)
+{
+    switch (fdwCtrlType)
+    {
+        // Handle the CTRL-C signal. 
+    case CTRL_C_EVENT:
+        // CTRL-CLOSE: confirm that the user wants to exit. 
+    case CTRL_CLOSE_EVENT:
+    case CTRL_BREAK_EVENT:
+    case CTRL_LOGOFF_EVENT:
+    case CTRL_SHUTDOWN_EVENT:
+        quit = true;
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+#endif
+
+void prevParam(void * /*userdata*/, const OSVR_TimeValue * /*timestamp*/,
+  const OSVR_ButtonReport *report)
+{
+  if (report->state == 1) {
+    activeParam--;
+    if (activeParam < 0) {
+      activeParam = static_cast<int>(params.size()) - 1;
+    }
+  }
+}
+
+void nextParam(void * /*userdata*/, const OSVR_TimeValue * /*timestamp*/,
+  const OSVR_ButtonReport *report)
+{
+  if (report->state == 1) {
+    activeParam++;
+    if (activeParam >= params.size()) {
+      activeParam = 0;
+    }
+  }
+}
+
+void setParams(void *userdata, const OSVR_TimeValue * /*timestamp*/,
+    const OSVR_ButtonReport *report)
+{
+  OSVRDisplayConfiguration *displayConfiguration = 
+    reinterpret_cast<OSVRDisplayConfiguration *>(userdata);
+
+  if (report->state == 1) {
+
+    // Create a new set of distortion parameters that has the
+    // specified parameters, but using the center of projection
+    // from the read-in values.
+    std::cout << "XXX New parameters: ";
+    for (size_t i = 0; i < params.size(); i++) {
+      std::cout << params[i] << " ";
```

Comment:
The comment says that it creates new set of params, but this block just prints them out. Is this just left when you were debugging?